tapeagents/llms.py

```diff
@@ -667,6 +666,8 @@ def batch_generate(self, prompts: list[Prompt]) -> list[LLMCall]:
             self.tokenizer.apply_chat_template(p.messages, add_special_tokens=True, add_generation_prompt=True)
             for p in prompts
         ]
+        if self.max_prompt_length > 0:
```

Comment:
I don't think we want this in TapeAgents, it's better to have prompt trimming in the downstream example code.